### PR TITLE
Add auth config

### PR DESCRIPTION
**Pull request by Random-Liu**
Fixes https://github.com/containerd/cri/issues/723.
Fixes https://github.com/containerd/cri/issues/835.

/cc @dmcgowan for the resolver change.

This PR:
1) Change `cri` plugin to use containerd docker resolver instead of our own fork. This gets rid of a lot of code duplication and maintenance overhead.
2) Make default registry credential configurable in the config file.

I'll mark this 1.11 and cherrypick into 1.11. Containerd 1.2 is our real release, I think it is fine to add new features before that.

### CODE REVIEW

File: docs/registry.md [hunk 1]
```diff
@@ -20,3 +21,23 @@ The endpoint is a list that can contain multiple image registry URLs split by co
...
+  [plugins.cri.registry.auths."https://gcr.io"]
+    username = ""
+    password = ""
+    auth = ""
+    identitytoken = ""
+```
+The meaning of each field is the same with the corresponding field in `.docker/config.json`.
+
+Please note that auth config passed by CRI takes precedence. The registry
+credential in the config file will only be used when auth config is not specified
```
Comment (mikebrow) [inline docs/registry.md]: /s/the config file/this config/

File: docs/registry.md [hunk 2]
```diff
@@ -20,3 +21,23 @@ The endpoint is a list that can contain multiple image registry URLs split by co
...
+[plugins.cri.registry.auths]
+  [plugins.cri.registry.auths."https://gcr.io"]
+    username = ""
+    password = ""
+    auth = ""
+    identitytoken = ""
+```
+The meaning of each field is the same with the corresponding field in `.docker/config.json`.
+
+Please note that auth config passed by CRI takes precedence. The registry
```
Comment (mikebrow) [inline docs/registry.md]: takes precedence over this config.

File: docs/registry.md [hunk 3]
```diff
@@ -20,3 +21,23 @@ The endpoint is a list that can contain multiple image registry URLs split by co
...
+credential in the config file will only be used when auth config is not specified
+by CRI.
```
Comment (mikebrow) [inline docs/registry.md]: /s/by CRI/by kubernetes via CRI/
Follow this [link](https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/) to read about configuring kubernetes pod to use images from private registries.

File: docs/registry.md (hunk 1 above)
Comment (Random-Liu) [reply docs/registry.md]: Done.

File: docs/registry.md (hunk 2 above)
Comment (Random-Liu) [reply docs/registry.md]: Done.

File: docs/registry.md (hunk 3 above)
Comment (Random-Liu) [reply docs/registry.md]: Done.

**Comment by Random-Liu** [comment]
Just updated containerd version to use https://github.com/containerd/cri/issues/835.

Re-apply LGTM.

**Pull request merged**